### PR TITLE
Move fiber init functions into wasm

### PR DESCRIPTION
**Pull request by Akaricchi**
`emscripten_fiber_init` and `emscripten_fiber_init_from_current_context` are trivial and do not interact with JS at all, so let them live in WASM

### CODE REVIEW

**Review by sbc100 — COMMENTED** [review]
I'm a huge fan of moving stuff to native code, but I don't think we need/want a new library here.

Can you just add this file to libc in the same way that we have `emscripten_console.c` is added on line 1133 of tools/system_libs.py.   You can also put the new file there in `system/lib/libc`

Otherwise LGTM!

**Comment by Akaricchi** [comment]
I can do that, but there are some subtle implications that may or may not be an issue:

1. By separating it into a library I made it only get linked when `ASYNCIFY` is 1 (which is the only mode where fibers are supported). Thus code that accidentally tries to use these functions, or forgets to link with `-s ASYNCIFY`, or tries to link with `-s ASYNCIFY=2`, will get rejected with a linker error.
I think this is a useful property, although I can see an argument against it for hypothetical code that wants to detect Asyncify availability at runtime with `emscripten_has_asyncify()`. I haven't seen any such code (that is also using fibers) in the wild, but as a workaround we can allow linking to `libasyncifyfiber` explicitly for this purpose (I guess the `can_use()` override has to be removed then?).
2. `libasyncifyfiber` is subject to LTO, while `libc` is not, as far as I understand. Not super important but can be a small nice bonus. These functions are small and probably will be inlined most of the time.

With that in mind, let me know what you want me to do. I'm not really against moving it into libc if you think that's a better idea.

P.S. I have no idea what the CI failures are about, probably not related to the changes?

**Comment by sbc100** [comment]
> **Edit**: I just realized that the availability of those symbols can probably be controlled via `REQUIRED_EXPORTS` independent of whether the functions are in libc or a separate library. If that's the case then the first point is moot.

Right, and libc is actually compiled as LTO .. at least any files that are not part of `self.non_lto_files`.   So I think its fine to put them in libc for now.  We can always change our minds later.

Putting the new file in libc will also solve the test failures related to `FROZEN_CACHE`

File: emcc.py [hunk 1]
```diff
@@ -2663,6 +2663,9 @@ def get_full_import_name(name):
 
     if settings.ASYNCIFY == 2:
       diagnostics.warning('experimental', 'ASYNCIFY with stack switching is experimental')
+    else:
+      settings.REQUIRED_EXPORTS += ['emscripten_fiber_init',
+                                    'emscripten_fiber_init_from_current_context']
```
Comment (sbc100) [inline emcc.py:2668]: Why do we need to export these two functions?  I don't see any usage of them in JS.. but maybe I'm missing something?

**Review by sbc100 — COMMENTED** [review]
lgtm % comment

File: emcc.py (hunk 1 above)
Comment (Akaricchi) [reply emcc.py:2668]: Previously I was getting link errors that suggested I should export them, despite them only being used in C code. That was when they still were in a separate library. I've just confirmed that it's not actually needed, at least not anymore. It probably wasn't needed back then either and I just got confused.

**Comment by Akaricchi** [comment]
Comment addressed. Do you want me to squash the changes?

**Review by sbc100 — APPROVED** [review]

**Pull request merged**